x/auth/types/account_retriever_test.go

```diff
@@ -1,44 +1,34 @@
 package types_test
 
 import (
-	"errors"
-	"fmt"
 	"testing"
 
-	"github.com/golang/mock/gomock"
 	"github.com/stretchr/testify/require"
 
-	"github.com/cosmos/cosmos-sdk/tests/mocks"
+	"github.com/cosmos/cosmos-sdk/testutil/network"
 	"github.com/cosmos/cosmos-sdk/x/auth/types"
 )
 
-var errFoo = errors.New("dummy")
-
 func TestAccountRetriever(t *testing.T) {
```

Comment:
@amaurymartiny I'm getting the following error:

```shell
        	Error Trace:	account_retriever_test.go:29
        	Error:      	Received unexpected error:
        	            	unexpected 'x-cosmos-block-height' header length; got 0, expected: 1
        	Test:       	TestAccountRetriever
```

What am I missing? AFAU, the header should always be added..

Reply:
This is related to https://github.com/cosmos/cosmos-sdk/pull/7006#discussion_r468718941, I only added this header for the gRPC server, not for requests coming from clientCtx.

Reply:
Got it. Yeah, we're gonna 100% need this. I'll hold off on the PR until we get that in. Thanks!